### DISCUSSION TITLE
Chart tick mark labels blurry

### DISCUSSION THREAD

**Discussion by webITSol** [primefaces/community#1832] [PrimeFaces] [answered]
I am maintaining a legacy application that uses primefaces 6.2. Some months ago the charts that come with primefaces started to produce rendering failures like the one in the attached export file. (See blurry tick mark labels.)
I was not able to reproduce the error reliably, because it occurs only sometimes on some machines (browsers?). Maybe there is some race condition that was introduced with a browser update?
Any idea how this can be fixed? Does the payed version come with the same problem?
![Download](https://github.com/primefaces/community/assets/17412136/a2bdbdbf-040a-4e80-8d58-76e5171477bc)

**Comment by melloware** [comment]
That is the old jquery JQPlot which is deprecated, no longer updated and long since been removed in PrimeFaces.   I don't think you will find much support for it here.  Weird though if its 6.2 with no updates that it started getting blurry now?   Did you make any other library changes?  Other than that I have to assume its sometihg with browser  and Javascript that changed recently?

Wonder if its related to this: https://stackoverflow.com/questions/20221461/hidpi-retina-plot-drawing/20228300#20228300

**Reply by webITSol** [reply]
Thanks for your reply! No, there were no library changes whatsoever. I do not think the issue is related to the canvas resolution. It would also be difficult to try it out, since all the jqPlot JS files are bundled with Primefaces.
Calling plot.replot() several times usually solves the issue...

**Reply by melloware** [reply]
Yeah most likely an old bug in jqPlot.  No updates in 6 years: https://github.com/jqPlot/jqPlot  even if you stay on 6.2 you might want to consider switching to Apache EChart or Chart.JS for those charts 😄

**Reply by melloware** [reply]
Oh check this out: https://github.com/primefaces/primefaces/issues/4655  You may have to apply that patch for your jqPlot..

**Reply by webITSol** [reply]
Thank you!

**Reply by melloware** [reply]
The other thing you can do is grab the last good "patched" version from PrimeFaces 11 found here: https://github.com/primefaces/primefaces/tree/11.0.0/primefaces/src/main/resources/META-INF/resources/primefaces/charts

And jam that into your PrimeFaces 6.2 JAR or patch a 6.2.1 JAR manually with the jqPlot that already has a bunch of fixes applied.

**Reply by webITSol** [reply]
That's a cool idea!

**Reply by melloware** [reply]
Let me know if it fixes your blurriness issues or not.

**Reply by webITSol** [reply]
Patching the .jar file with the most recent chart.js file works. You just have to rename the 0-chartjs.min.js file to charts.js.
But it does not fix the problem. We'll probably have to upgrade to another chart framework. Maybe we can integrate google charts without too much effort.

**Reply by melloware** [reply, marked as answer]
I recommend Apache ECharts  Super easy and amazing charts: https://echarts.apache.org/examples/en/index.html
